### PR TITLE
Cpdrp 283 manage schools cohort

### PR DESCRIPTION
**Pull request by tonyheadford**
### Context
[Jira](https://dfedigital.atlassian.net/jira/software/projects/CPDRP/boards/102?selectedIssue=CPDRP-283)
View a school's induction choice by cohort

### Changes proposed in this pull request
Adds view of cohorts to admin pages

### Testing
I have added cohorts and induction choices for 2022 so that we can test more than 1 cohort. The schools in the test data with URNs that are greater than "000100" and are even numbers, have choices for both 2021 and 2022.

### Review Checks
- [ ] All pages have automated accessibility checks via cypress
- [ ] All pages have visual tests via cypress + percy
- [ ] All `School` queries are correctly scoped - `eligible` when they need to be

### How can I view this in a review app?
Given you are signed in as a DfE Admin user, select a school from the list of schools and then select "Cohorts" from the nav menu. You should see the induction for each cohort.

### CODE REVIEW

File: app/policies/school_cohort_policy.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # frozen_string_literal: true
 
 class SchoolCohortPolicy < ApplicationPolicy
+  def index?
+    user.admin?
+  end
```
Comment (kicferk1) [inline app/policies/school_cohort_policy.rb:6]: Can we have a test for it?

File: app/models/school_cohort.rb [hunk 2]
```diff
@@ -18,6 +18,14 @@ def training_provider_status
     school.partnerships&.active&.exists?(cohort: cohort) ? "Done" : "To do"
   end
 
+  def lead_provider
+    school.lead_provider(cohort.start_year)
+  end
+
+  def delivery_partner
+    school.delivery_partner_for(cohort.start_year)
+  end
```
Comment (kicferk1) [inline app/models/school_cohort.rb:27]: Can we unit test those?

File: app/controllers/admin/schools/cohorts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    skip_after_action :verify_authorized
+    before_action :set_school
+
+    def index
+      @school_cohorts = policy_scope(SchoolCohort)
+        .joins(:school)
+        .where(school: @school)
+        .joins(:cohort)
+        .order("cohorts.start_year ASC")
+    end
```
Comment (kicferk1) [inline app/controllers/admin/schools/cohorts_controller.rb:14]: Do we need to paginate it? There will be thousands of schools with a school cohort :/

Comment (kicferk1) [reply app/controllers/admin/schools/cohorts_controller.rb:14]: I guess the second part is: Do we need this list to be searchable? Not necessarily in this PR, just whether it's taken into account in general.

Comment (tonyheadford) [reply app/controllers/admin/schools/cohorts_controller.rb:14]: This is only the cohorts for a single school?

File: db/seeds/test_data.rb [hunk 4]
```diff
@@ -1,6 +1,7 @@
 # frozen_string_literal: true
 
 DOMAIN = "@digital.education.gov.uk" # Prevent low effort email scraping
+cohort_2022 = Cohort.find_or_create_by!(start_year: 2022)
```
[the remark this answers is not in the document]
Comment (tonyheadford) [reply db/seeds/test_data.rb:4]: Maybe, I don't know the reason why this was split into initial and test seeds so put it in the test seeds - should that future cohort go in the initial seeds now?

File: app/policies/school_cohort_policy.rb (hunk 1 above)
Comment (tonyheadford) [reply app/policies/school_cohort_policy.rb:6]: Good spot, actually this is not needed now so I will remove it.

File: db/seeds/test_data.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (kicferk1) [reply db/seeds/test_data.rb:4]: So `initial_seed` is the common part between all environments. The other seed files are for more specific envs, with fake data that we don't want on 'real' versions.

I don't think we need the 2022 cohort in the initial seed file any time soon, but it is a piece of data that is correct and likely the same across all environments. Thoughts @ethanmills ?

**Review by kicferk1 — APPROVED** [review]

File: db/seeds/test_data.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethanmills) [reply db/seeds/test_data.rb:4]: Let's leave it here until we've thought about dealing with multiple cohorts properly. We don't need it any time soon on production, and I can't see any benefit in having it there atm

**Review by ethanmills — APPROVED** [review]

**Review by jamesrose — APPROVED** [review]

**Pull request merged**